### DISCUSSION TITLE
Criteria for adding additional functions?

### DISCUSSION THREAD

**Discussion by Maushundb** [char0n/ramda-adjunct#1721] [General]
@char0n Do we have a policy or set of guidelines for what functions will and will not be added to the library? I imagine we want to avoid bloat if the function is too niche or similar to something existing, but also want to make it easy for folks to know if their contribution will likely be accepted.

**Comment by char0n** [comment]
@Maushundb everybody has a possibility to create an issue with his new feature proposal and elaborate on it there. Usually there is a discussion about it and if consensus is made we push it forward by properly labeling it. 

This repo is governed in [BNLF mode](https://github.com/char0n/ramda-adjunct/blob/master/GOVERNANCE.md#benevolent-dictator-project-lead), which means that even when there is a consensus by contributors or community I can still stop it getting in the codebase if I don't consider it suitable addition to the library. Example of this can be [ramda-log](https://www.npmjs.com/package/ramda-log) which lot of users requested, but I did not allow it to become part of the codebase (obvious impurity + side effects).


### Objective parameters of new features:

1.) new features needs to be pure (one exception is various promise transformers and utils)
2.) cannot be toooo specific (this means having feature specific to one person's usecase)
3.) must solve a real problem

**Comment by Maushundb** [comment]
Got it, that makes sense. Thanks!
